### PR TITLE
Make axj outputting the same data as ax

### PR DESCRIPTION
**Pull request by wargio**
`ax` output
```
                           segment.LOAD0 0x8048000 ->      CODE -> 0x8048047 segment.PHDR+19
                       section..dynstr+8 0x8048278 ->      CODE -> 0x80482ee section..gnu.version_r+10
                      section..dynstr+11 0x804827b ->      CODE -> 0x80482f1 section..gnu.version_r+13
                      section..dynstr+21 0x8048285 ->      CODE -> 0x80482f6 section..gnu.version_r+18
                      section..dynstr+50 0x80482a2 ->      CODE -> 0x8048308 section..rel.dyn+4
                      section..dynstr+70 0x80482b6 ->      CODE -> 0x804832c section..rel.plt+32
                         section..init+6 0x8048342 ->      CALL -> 0x80483f4 fcn.080483f4
                        section..init+11 0x8048347 ->      CALL -> 0x8048450 sym.frame_dummy
                        section..init+16 0x804834c ->      CALL -> 0x8048620 sym.__do_global_ctors_aux
                          section..plt+6 0x804835a ->      CODE -> 0x8049ffc section..got.plt+8
               sym.imp.__libc_start_main 0x8048364 ->      CODE -> 0x804a000 reloc.__libc_start_main
            sym.imp.__libc_start_main+11 0x804836f ->      CODE -> 0x8048354 section..plt
                           sym.imp.scanf 0x8048374 ->      CODE -> 0x804a004 reloc.scanf
                        sym.imp.scanf+11 0x804837f ->      CODE -> 0x8048354 section..plt
                          sym.imp.strlen 0x8048384 ->      CODE -> 0x804a008 reloc.strlen
                       sym.imp.strlen+11 0x804838f ->      CODE -> 0x8048354 section..plt
                          sym.imp.printf 0x8048394 ->      CODE -> 0x804a00c reloc.printf

```
`axj` output
```
[{
    "name": "segment.LOAD0",
    "at": 134512640,
    "type": "CODE",
    "address": 134512711,
    "xref": "segment.PHDR+19"
}, {
    "name": "section..dynstr+8",
    "at": 134513272,
    "type": "CODE",
    "address": 134513390,
    "xref": "section..gnu.version_r+10"
}, {
    "name": "section..dynstr+11",
    "at": 134513275,
    "type": "CODE",
    "address": 134513393,
    "xref": "section..gnu.version_r+13"
}, {
    "name": "section..dynstr+21",
    "at": 134513285,
    "type": "CODE",
    "address": 134513398,
    "xref": "section..gnu.version_r+18"
}, {
    "name": "section..dynstr+50",
    "at": 134513314,
    "type": "CODE",
    "address": 134513416,
    "xref": "section..rel.dyn+4"
}, {
    "name": "section..dynstr+70",
    "at": 134513334,
    "type": "CODE",
    "address": 134513452,
    "xref": "section..rel.plt+32"
}, {
```

### CODE REVIEW

File: libr/anal/xrefs.c [hunk 1]
```diff
@@ -251,15 +251,24 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
 				} else {
 					anal->cb_printf (",");
 				}
-				anal->cb_printf ("\"%"PFMT64d"\":%"PFMT64d, ref->at, ref->addr);
+
```
Comment (radare) [inline libr/anal/xrefs.c]: Check for null

File: libr/anal/xrefs.c [hunk 2]
```diff
@@ -251,15 +251,24 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
...
+				char *name = anal->coreb.getNameDelta (anal->coreb.core, ref->at);
+				r_str_replace_char (name, ' ', 0);
+				anal->cb_printf ("{\"name\":\"%s\",", name? name: "");
+				free (name);
+				anal->cb_printf ("\"at\":%"PFMT64d",\"type\":\"%s\",\"address\":%"PFMT64d, ref->at, r_anal_xrefs_type_tostring (t), ref->addr);
+				name = anal->coreb.getNameDelta (anal->coreb.core, ref->addr);
+				r_str_replace_char (name, ' ', 0);
+				anal->cb_printf (",\"xref\":\"%s\"}", (name && *name) ? name : "");
```
Comment (radare) [inline libr/anal/xrefs.c]: You are checking for name&&*name in 2 places in different ways. Maybe better to use r_str_get to unify behaviours

File: libr/anal/xrefs.c (hunk 1 above)
Comment (wargio) [reply libr/anal/xrefs.c]: where?

File: libr/anal/xrefs.c [hunk 3]
```diff
@@ -215,29 +215,24 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
...
 		int t = ref->type ? ref->type: ' ';
 		switch (rad) {
 		case '*':
-			anal->cb_printf ("ax%c 0x%"PFMT64x" 0x%"PFMT64x"\n",
-				t, ref->addr, ref->at);
+			anal->cb_printf ("ax%c 0x%"PFMT64x" 0x%"PFMT64x"\n", t, ref->addr, ref->at);
 			break;
 		case '\0':
 			{
 				char *name = anal->coreb.getNameDelta (anal->coreb.core, ref->at);
```
Comment (radare) [inline libr/anal/xrefs.c]: Check for null

Comment (wargio) [reply libr/anal/xrefs.c]: i don't think it is really needed. all the functions, that use it, checks for null input.

File: libr/anal/xrefs.c [hunk 4]
```diff
@@ -215,30 +215,33 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
...
-				r_str_replace_char (name, ' ', 0);
-				anal->cb_printf ("%40s", name? name: "");
-				free (name);
+				if (name) {
+					r_str_replace_ch (name, ' ', 0, true);
+					anal->cb_printf ("%40s", name);
+					free (name);
+				} else {
+					anal->cb_printf ("%40s", "(null)");
```
Comment (radare) [inline libr/anal/xrefs.c]: wtf why are you printing null with 40 spaces of padding?? i would prefer to not show anything in that case, (null). i mean if you want to show null just use r_str_get.. but i think this is getting overengineered

Comment (wargio) [reply libr/anal/xrefs.c]: because i didn't wanted to change the behavior of the code. before it was displaying empty line with 40 spaced chars.

**Comment by radare** [comment]
Using ??? Or *noname* or nothing (not even spaces) will be better than showing (null) which will look like a bug to the user

**Comment by wargio** [comment]
ok.

File: libr/anal/xrefs.c [hunk 5]
```diff
@@ -251,15 +254,32 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
...
+					anal->cb_printf ("{\"name\":\"\",");
+				}
+				anal->cb_printf ("\"from\":%"PFMT64d",\"type\":\"%s\",\"addr\":%"PFMT64d, ref->at, r_anal_xrefs_type_tostring (t), ref->addr);
+				name = anal->coreb.getNameDelta (anal->coreb.core, ref->addr);
+				if (name) {
+					r_str_replace_ch (name, ' ', 0, true);
+					anal->cb_printf (",\"xref\":\"%s\"}", name);
+					free (name);
+				} else {
+					anal->cb_printf (",\"xref\":\"\"}");
```
[the remark this answers is not in the document]
Comment (wargio) [reply libr/anal/xrefs.c]: i need it anyway. i have to end the json object.

File: libr/anal/xrefs.c [hunk 6]
```diff
@@ -215,30 +215,33 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
...
 				char *name = anal->coreb.getNameDelta (anal->coreb.core, ref->at);
-				r_str_replace_char (name, ' ', 0);
-				anal->cb_printf ("%40s", name? name: "");
-				free (name);
+				if (name) {
+					r_str_replace_ch (name, ' ', 0, true);
+					anal->cb_printf ("%40s", name);
+					free (name);
+				} else {
+					anal->cb_printf ("%40s", "noname");
```
[the remark this answers is not in the document]
Comment (wargio) [reply libr/anal/xrefs.c]: things will not be aligned.

File: libr/anal/xrefs.c [hunk 7]
```diff
@@ -251,15 +254,32 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
...
 				}
-				anal->cb_printf ("\"%"PFMT64d"\":%"PFMT64d, ref->at, ref->addr);
+
+				char *name = anal->coreb.getNameDelta (anal->coreb.core, ref->at);
+				if (name) {
+					r_str_replace_ch (name, ' ', 0, true);
+					anal->cb_printf ("{\"name\":\"%s\",", name);
+					free (name);
+				} else {
+					anal->cb_printf ("{\"name\":\"\",");
```
[the remark this answers is not in the document]
Comment (wargio) [reply libr/anal/xrefs.c]: else branch is needed to start the json.

File: libr/anal/xrefs.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (radare) [reply libr/anal/xrefs.c]: just keep the 40 spaces with an empty string, or use "???" which is less likely to be a valid name for anything for example or just one ? :D

**Pull request merged**